pkg/api/api.go

```diff
@@ -21,17 +22,24 @@ func New(profileCatalog *catalog.Catalog) API {
 	}
 
 	r.HandleFunc("/profiles", a.ProfilesHandler)
+	r.HandleFunc("/profiles/{name}", a.ProfileHandler)
+
 	return a
 }
 
 func (a *API) ProfilesHandler(w http.ResponseWriter, r *http.Request) {
-	profileName := r.URL.Query().Get("name")
-	out, err := json.Marshal(a.profileCatalog.Search(profileName))
-	if err != nil {
-		panic(err)
-	}
-	_, err = w.Write(out)
-	if err != nil {
-		panic(err)
+	query := r.URL.Query().Get("name")
+	marshalResponse(w, a.profileCatalog.Search(query))
+}
+
+func (a *API) ProfileHandler(w http.ResponseWriter, r *http.Request) {
+	profileName := mux.Vars(r)["name"]
+	marshalResponse(w, a.profileCatalog.Get(profileName))
+}
+
+func marshalResponse(w http.ResponseWriter, v interface{}) {
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(v); err != nil {
+		log.Printf("failed to encode response: %s", err)
```

Comment:
shall we change the return code if we fail to encode the response?

Reply:
🤔  do you mean like:

```go
w.Header().Set("Content-Type", "application/json")
encoder := json.NewEncoder(w)
if err := encoder.Encode(v); err != nil {
  w.WriteHeader(500) // whatever
  if err := encoder.Encode(someErrorTypeThing); err != nil {
    log.Printf("failed to encode error setting thing: %s", err)
  }
  log.Printf("failed to encode response: %s", err)
}
```

i feel logging should be enough, not sure if failing to encode is a thing which happens easily

Reply:
I just mean adding the `w.WriteHeader(500)`. otherwise we return a `200` even though we've failed to write the response, producing a weird behaviour for the consumer of the API

Reply:
oh so just like: 

```go
w.Header().Set("Content-Type", "application/json")
if err := json.NewEncoder(w).Encode(v); err != nil {
  w.WriteHeader(500)
  log.Printf("failed to encode response: %s", err)
}
```

Reply:
yeah